Blog - Why we love GitHub as our CMS (#5486)
PostHog/posthog.com: contents/blog/github-cms.md

---
title: Why we use GitHub as our CMS

> date: 2023-03-14

Reviewer: ```suggestion
date: 2023-03-23
```

author: ["ian-vanagas"]
showTitle: true
rootpage: /blog
sidebar: Blog
hideAnchor: true
featuredImage: ../images/blog/posthog-blog-image.png
featuredImageType: full
category: Inside PostHog
tags:
 - Marketing
---

> As the amount of content you create grows, inevitably you need a way to manage it. A content management system, or CMS, is the go-to software solution for this. Examples include WordPress, Drupal, Wix, Ghost, and many more. 

Reviewer: ```suggestion
As you publish more content, inevitably you need a way to manage it. A content management system, or CMS, is the go-to software solution for this. Examples include WordPress, Drupal, Wix, Ghost, and many more. 
```

> But there's left field option that doesn’t get thought of as a CMS: GitHub. In this post, I'll explain the three reasons we use GitHub as a CMS (and love doing so). 

Reviewer: ```suggestion
But there's another left-field option: GitHub. In this post, I'll explain the three reasons we use GitHub as a CMS (and love doing so). 
```

## 1. GitHub enables transparency and open contributions

[Being open source](https://github.com/PostHog/posthog.com) is core to PostHog. It creates the transparency, context, and trust needed to be a high-performing, remote software startup.

As the home of open source, GitHub makes this easy. Having our level of transparency might not be possible with another CMS. Anyone can see what we're working on, give feedback, and even make changes to the content and code. This happens all the time, for example:

- Engineers contribute to site documentation and review content.
- Marketers fix layout or template issues.
- Readers fix typos and content issues, this makes up ~5% of merged pull requests on our [posthog.com repo](https://github.com/PostHog/posthog.com).
- Users provide feedback on [our roadmap](/roadmap) which displays data from GitHub issues on our site.

Every time one of these happens, it makes our site a bit better in a way that wouldn’t be possible if we weren’t using GitHub. It empowers people to make changes to code and content themselves, rather than rely on others to do it. This enables us to move fast and have a high impact while maintaining quality.

> ## GitHub has most of the necessary tools for your content workflow

Reviewer: ```suggestion
## 2. GitHub has most of the necessary tools for your content workflow
```

## 3. GitHub keeps us engineering-focused

At PostHog, [everybody codes](/handbook/company/values#everyone-codes), and our [ideal customers](/blog/creating-ideal-customer-profile) are developers. Using GitHub and combining content and code keeps this top of mind. GitHub is the ideal CMS for engineers and developers because they already use it, and it encourages non-technical people to be more technical.

Developers don’t want to spend time in a CMS. They want to use tools they already know and spend time in. Trying to get a busy developer to contribute on GitHub is easier than trying to get them to contribute in a CMS they aren’t familiar with. This means they make more changes to our site and review content more often. **~40% of [merged pull requests in the posthog.com repo](https://github.com/PostHog/posthog.com/pulls?q=is%3Aclosed) are from people outside of marketing, website, and docs.**

> It also encourages our marketing team to be more technical. They must understand how the code and content interact. For example, missing metadata causes an error in the web app. The marketing team can figure out and solve this themselves, rather than asking the development team to do it.

Reviewer: ```suggestion
It also encourages our marketing team to be more technical. They must understand how the code and content interact. For example, when missing metadata causes an error in the web app. The marketing team can figure out and solve this themselves, rather than asking the development team to do it.
```

---
title: Why we use GitHub as our CMS
date: 2023-03-14
author: ["ian-vanagas"]
showTitle: true
rootpage: /blog
sidebar: Blog
hideAnchor: true
featuredImage: ../images/blog/posthog-blog-image.png
featuredImageType: full
category: Inside PostHog
tags:
 - Marketing
---

> As the amount of content you create grows, inevitably you need some way to manage it. A content management system, or CMS, is the go-to software solution for this. Examples include WordPress, Drupal, Wix, Ghost, and many more.
> 
> One that doesn’t get thought of as a CMS is GitHub. We use GitHub and love it for three main reasons:
> 
> 1. Enables transparency and open contributions
> 2. Provides the necessary content creation workflow tools
> 3. Keeps us engineering-focused

Reviewer: ```suggestion
As the amount of content you create grows, inevitably you need some way to manage it. A content management system, or CMS, is the go-to software solution for this. Examples include WordPress, Drupal, Wix, Ghost, and many more. 

But there's left field option that doesn’t get thought of as a CMS: GitHub. In this post, I'll explain the three reasons we use GitHub as a CMS (and love doing so). 
```

On reflection, I think works better without the bullet points in the intro.

> ## Enables transparency and open contributions

Reviewer: ```suggestion
## 1. GitHub enables transparency and open contributions
```

2. Issues go into a content [project board](https://docs.github.com/en/issues/tracking-your-work-with-issues/planning-and-tracking-work-for-your-team-or-project#adding-issues-to-a-project-board) with columns for the backlog, coming soon (in priority order), in progress, in review, and done.

3. Content gets worked on. When done, the creator opens a pull request (with a [checklist](https://github.com/PostHog/posthog.com/blob/master/.github/pull_request_template.md)), and asks for reviews. [GitHub Actions](/blog/automating-a-software-company-with-github-actions) run for spellcheck and Vercel site previews.

4. Reviewers comment and make suggestions. Creator updates until approved.

5. Content gets merged and changes publish to the production site automatically.

GitHub and a text editor are all we need to go from idea to published content. The features of GitHub provide many benefits to our workflow such as:

- Automation for templates, checklists, spellcheck, review notifications, and site previews.

- Central location for ideas, drafts, team comments, finished content, and revisions.

- Tools for feedback including suggestions, full markdown comments, and comment history.

> The piece that is missing is a content calendar. The solution is to ship content as soon as it is ready. If the content relates to a launch, create a branch with all the changes and merge it as soon as the feature launches.

Reviewer: ```suggestion
The one thing it lacks is a content calendar, but it's not something we miss much. Most content can be published as soon as it's ready. If the content relates to a launch, we create a branch with all the changes (including documentation and landing page updates) and merge it as soon as the feature launches.
```

featuredImageType: full
category: Inside PostHog
tags:
 - Marketing
---

As the amount of content you create grows, inevitably you need some way to manage this. Content management systems, or CMS, are the go-to software solution for this. Examples include WordPress, Drupal, Wix, Ghost, and many more. 

One that doesn’t get thought of as a CMS is GitHub. We use GitHub and love it for three main reasons:

1. Stores content and code together
2. Provides the necessary content creation workflow tools
3. Keeps us engineering-focused

## Content and code together

Content management systems often disconnect content from code. Content lives in the CMS, and code somewhere else. Often, the code is somewhere the marketing team can’t even access it. For example, millions of people use WordPress without ever seeing (or changing) a single line of code.

This disconnects the content from the site, and the user experience is worse because of it. The development of the site and content should align to create a great user experience. Content disconnected from code causes issues. Using GitHub to store everything together prevents this disconnection.

> Keeping our content with our code forces both to work together. This prevents content from causing issues on the site. If a piece of content needs a new type of page, work on both to happen simultaneously and for them to launch at the same time. Launches can contain multiple pages, components, and pieces of content seamlessly. As a recent example, see the addition of our [PostHog for startups landing page](https://github.com/PostHog/posthog.com/pull/5451).

Author: Yup, rewrote.

![Startup landing page](../images/blog/github-cms/startups.png)

With GitHub as a CMS, we can use code, frameworks, and packages directly without having to rely on integrations with a CMS. We can get the latest updates and innovations from the tools we use to help our website improve. Instead of the effort going into maintaining a CMS connection, it goes into making the site better.

Having code and content in one place enables us to open source. This is a [key value for us](/handbook/company/values#we-are-open-source). Among other things, it forces us to "work with the door open" and allow anyone to contribute. For example:

- Engineers can contribute to site documentation.
- Marketers can fix layout or template issues.
- Readers can fix typos and make suggestions, this makes up ~5% of merged pull requests on our [posthog.com repo](https://github.com/PostHog/posthog.com).

> Every time one of these changes happens, it makes our site a bit better in a way that wouldn’t be possible if we weren’t using GitHub. 

Author: I rewrote this section for this to be the key point.

Developers don’t want to spend time in a CMS. They want to use tools they already know and spend time in. Trying to get a busy developer to contribute on GitHub is easier than trying to get them to contribute in a CMS they aren’t familiar with. This means they make more changes to our site and review content more often. **~40% of [merged pull requests in the posthog.com repo](https://github.com/PostHog/posthog.com/pulls?q=is%3Aclosed) are from people outside of marketing, website, and docs.**

It also encourages our marketing team to be more technical. They must understand how the code and content interact. For example, missing metadata causes an error in the web app. The marketing team can figure out and solve this themselves, rather than asking the development team to do it.

![VSCode screenshot of this blog](../images/blog/github-cms/blog.png)

By using GitHub as a CMS, we stay engineering-focused. Everyone works like an engineer, and that enables us to understand and build a better product for them. This helps PostHog succeed in the long run. 

> If your team values engineering focus, adopting GitHub as a CMS is a practical way to make it happen.

Reviewer: This is the most convincing section + ends on a great note.
